src/movies.js

```diff
@@ -1,16 +1,107 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+  return movies.map(function (movie) {
+    return movie.director;
+  });
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+function howManyMovies(movies) {
+  let count = 0;
+  if (movies.length === 0) return 0;
+  for (let movie of movies) {
+    if (movie.director.includes("Spielberg") && movie.genre.includes("Drama"))
+      count++;
+  }
+  return count;
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
-
+function ratesAverage(movies) {
+  if (movies.length === 0) return 0;
+  return parseFloat(
+    (
+      movies.reduce(function (acc, val) {
+        return acc + (val.rate || 0);
+      }, 0) / movies.length
+    ).toFixed(2)
+  );
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(movies) {
+  let noDrama = true;
+  let count = 0;
+  let rateSum = 0;
+  movies.reduce(function (acc, val) {
+    if (val.genre.includes("Drama")) {
+      rateSum += val.rate;
+      count++;
+      noDrama = false;
+    }
+  }, 0);
+  if (noDrama) return 0;
+  return parseFloat((rateSum / count).toFixed(2));
+}
```

Comment:
This is nice solution!
For a shorter solution, you can filter the movies that include the Drama genre and then past this new array as an argument to the ratesAverage(filteredMoviesArr) function. ;)

Reply:
That's an elegant solution.
Thanks for the review and the helpful comments :D